ethereum/rpc/namespaces/eth/api.go

```diff
@@ -218,7 +229,18 @@ func (e *PublicAPI) GetBalance(address common.Address, blockNum rpctypes.BlockNu
 }
 
 // GetStorageAt returns the contract storage at the given address, block number, and key.
-func (e *PublicAPI) GetStorageAt(address common.Address, key string, blockNum rpctypes.BlockNumber) (hexutil.Bytes, error) { // nolint: interfacer
+func (e *PublicAPI) GetStorageAt(address common.Address, key string, blockNrOrHash rpctypes.BlockNumberOrHash) (hexutil.Bytes, error) { // nolint: interfacer
+	var blockNum rpctypes.BlockNumber
+	if blockNrOrHash.BlockHash != nil {
+		blockHeader, err := e.backend.HeaderByHash(*blockNrOrHash.BlockHash)
+		if err != nil {
+			return nil, err
+		}
+		blockNum = rpctypes.NewBlockNumber(blockHeader.Number)
+	} else if blockNrOrHash.BlockNumber != nil {
+		blockNum = *blockNrOrHash.BlockNumber
+	}
+
 	e.logger.Debug("eth_getStorageAt", "address", address.Hex(), "key", key, "block number", blockNum)
```

Comment:
I'd define this before the blockNum statement you defined

Reply:
fixed